src/test/app/LendingHelpers_test.cpp

```diff
@@ -620,10 +620,117 @@ class LendingHelpers_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testTryOverpaymentValueChange()
+    {
+        // This test ensures that overpayment value change is computed
+        // correctly.
+        testcase("tryOverpayment - Value Change is the decrease in interest");
+
+        using namespace jtx;
+        using namespace ripple::detail;
+
+        Env env{*this};
+        Account const issuer{"issuer"};
+        PrettyAsset const asset = issuer["USD"];
+
+        // Interest delta is 40 (100 - 50 - 10)
+        ExtendedPaymentComponents const overpaymentComponents = {
+            PaymentComponents{
+                .trackedValueDelta = Number{50, 0},
+                .trackedPrincipalDelta = Number{50, 0},
+                .trackedManagementFeeDelta = Number{0, 0},
+                .specialCase = PaymentSpecialCase::extra,
+            },
+            numZero,
+            numZero,
+        };
+
+        TenthBips16 managementFeeRate{20'000};  // 10%
+        TenthBips32 loanInterestRate{10'000};   // 20%
+        Number loanPrincipal{1'000};
+        std::uint32_t paymentInterval = 30 * 24 * 60 * 60;
+        std::uint32_t paymentsRemaining = 10;
+        std::int32_t loanScale = -5;
+        auto const periodicRate =
+            loanPeriodicRate(loanInterestRate, paymentInterval);
+
+        auto loanProperites = computeLoanProperties(
+            asset,
+            loanPrincipal,
+            loanInterestRate,
+            paymentInterval,
+            paymentsRemaining,
+            managementFeeRate,
+            loanScale);
+        std::cout << loanProperites.periodicPayment << std::endl;
+        std::cout << loanProperites.loanState.valueOutstanding << std::endl;
+        std::cout << loanProperites.loanState.interestOutstanding()
+                  << std::endl;
```

Comment:
Please change `std::cout` to `log` so the messages can be suppressed at runtime if the caller doesn't want to see them.